botocore/docs/utils.py

```diff
@@ -0,0 +1,614 @@
+import inspect
+from collections import namedtuple
+
+
+def py_type_name(type_name):
+    """
+    Get the Python type name for a given model type.
+
+        >>> py_type_name('list')
+        'list'
+        >>> py_type_name('structure')
+        'dict'
+
+    :rtype: string
+    """
+    return {
+        'blob': 'bytes',
+        'character': 'string',
+        'double': 'float',
+        'long': 'integer',
+        'map': 'dict',
+        'structure': 'dict',
+        'timestamp': 'datetime',
+    }.get(type_name, type_name)
+
+
+def py_default(type_name):
+    """
+    Get the Python default value for a given model type, useful
+    for generated examples.
+
+        >>> py_default('string')
+        '\'string\''
+        >>> py_default('list')
+        '[...]'
+        >>> py_default('unknown')
+        '...'
+
+    :rtype: string
+    """
+    return {
+        'double': '123.0',
+        'long': '123',
+        'integer': '123',
+        'string': "'string'",
+        'blob': "b'bytes'",
+        'boolean': 'True|False',
+        'list': '[...]',
+        'map': '{...}',
+        'structure': '{...}',
+        'timestamp': 'datetime(2015, 1, 1)',
+    }.get(type_name, '...')
+
+
+def get_official_service_name(service_model):
+    """Generate the official name of an AWS Service
+
+    :param service_model: The service model representing the service
+    """
+    official_name = service_model.metadata.get('serviceFullName')
+    short_name = service_model.metadata.get('serviceAbbreviation', '')
+    if short_name.startswith('Amazon'):
+        short_name = short_name[7:]
+    if short_name.startswith('AWS'):
+        short_name = short_name[4:]
+    if short_name and short_name.lower() not in official_name.lower():
+        official_name += ' ({0})'.format(short_name)
+    return official_name
+
+
+def get_instance_methods(instance):
+    """Retrieves an objects public methods and variables
+
+    :param instance: The instance of the class to inspect
+    :rtype: dict
+    :returns: A dictionary that represents an instance's methods where
+        the keys are the name of the methods and the
+        values are the handler to the method.
+    """
+    instance_members = inspect.getmembers(instance)
+    instance_methods = {}
+    for name, member in instance_members:
+        if not name.startswith('_'):
+            if inspect.ismethod(member):
+                instance_methods[name] = member
+    return instance_methods
+
+
+_DocumentedShape = namedtuple(
+    'DocumentedShape', ['name', 'type_name', 'documentation', 'metadata'])
+
+
+class DocumentedShape (_DocumentedShape):
+    """Use this class to inject new shapes into a model for documentation"""
+    def __new__(cls, name, type_name, documentation, metadata=None):
+        if metadata is None:
+            metadata = []
+        return super(DocumentedShape, cls).__new__(
+            cls, name, type_name, documentation, metadata)
+
+
+class BaseMethodSignatureDocumentor(object):
```

Comment:
These classes seem out of place in a utils module.  These classes have a lot of the core documentation logic.  Seems like they should be in their own module.